### PR TITLE
Add missing Cupertino widgets to the widget library

### PR DESCRIPTION
**Pull request by sfshaza2**
Fixes https://github.com/flutter/website/issues/8739

Adds 37 missing Cupertino widgets to the widget library.
@MitchellGoodwin will provide the missing images. THANKS, @MitchellGoodwin!

### CODE REVIEW

**Comment by sfshaza2** [comment]
Also, I tried viewing the staged version of this page and it was no longer available. I'd like to see that, once you do some restructuring.

File: src/_data/catalog/widgets.json [hunk 1]
```diff
@@ -625,12 +865,32 @@
     "categories": [
       "Cupertino (iOS-style widgets)"
     ],
+    "subcategories": [],
+    "link": "https://api.flutter.dev/flutter/cupertino/CupertinoSliverNavigationBar-class.html",
+    "image": "<img alt='' src='/assets/images/docs/catalog-widget-placeholder.png'>"
```
Comment (MitchellGoodwin) [inline src/_data/catalog/widgets.json:870]: Looks like this puts the placeholder on the CupertinoSliverNavigationBar, and the CupertinoSliverNavigationBar image on CupertinoSliverRefreshControl.

File: src/_data/catalog/widgets.json [hunk 2]
```diff
@@ -469,6 +519,16 @@
     "link": "https://api.flutter.dev/flutter/cupertino/CupertinoContextMenu-class.html",
     "image": "<img alt='' src='/assets/images/docs/widget-catalog/cupertino-context-menu.png'>"
   },
+  {
+    "name": "CupertinoContextMeniAction",
```
Comment (MitchellGoodwin) [inline src/_data/catalog/widgets.json:523]:
```suggestion
    "name": "CupertinoContextMenuAction",
```

File: src/_data/catalog/widgets.json [hunk 3]
```diff
@@ -681,6 +951,106 @@
...
+    "description": "iOS-style text selection controls.",
+    "categories": [
+      "Cupertino (iOS-style widgets)"
+    ],
+    "subcategories": [],
+    "link": "https://api.flutter.dev/flutter/cupertino/CupertinoTextSelectionControls-class.html",
+    "image": "<img alt='' src='/assets/images/docs/catalog-widget-placeholder.png'>"
+  },
+  {
+    "name": "CupertinoTextSelectionHandleControls",
```
Comment (MitchellGoodwin) [inline src/_data/catalog/widgets.json:985]: This has been deprecated. Probably shouldn't be included.

**Review by parlough — APPROVED** [review]
Nice! Thank you both for adding these entries and images to the catalog! Looks great.

**Pull request merged**